### ISSUE TITLE
'Auto Sign out' value ignored; Signs out after 5-10 mins during work

### ISSUE THREAD

**Issue by christianrr** [directus/app#2029]
## Bug Report
Although I set the 'Auto Sign out' to 300 minutes in global settings, the app kicks me out after approx. 5-10 minutes - just during work (this means that appearently there is also a problem with tracking inactivity). If I set the timeout to 0 or -1 minutes, the app also signs out after a couple of mins.

Did not happen before last update (Shield).

## Steps to reproduce

- Goto global settings, set 'Auto Sign out' to a arbitrary number, e.g. 300
- Work in collections
- App automatically signs out after approx. 5-10 minutes

## Technical Details
Device: [Desktop, Chrome 75.0.3770.142]
OS: [Windows 10 1809]
Web Server: [Apache 2.4.29]
PHP Version: [7.2.19]
Database: [5.7.27]
API Version: [2.5.0]
APP Version: [7.10.0]

**Comment by benhaynes** [comment]
@rijkvanzanten has another ticket (in the API) I believe, that should solve this. It is a different approach to storing auth that would allow users to stay logged in for much longer periods of time.

**Comment by rijkvanzanten** [comment]
Ref https://github.com/directus/api/pull/1280 and https://github.com/directus/api/issues/1263

**Comment by gurumark** [comment]
Would it be possible to create permanent login where users will be logged in as long as they don't sign out? This could be necessary to support blogging usecases.

**Comment by rijkvanzanten** [comment]
Please see the issue I referenced in my reply to your issue: https://github.com/directus/api/issues/1263 and the PR that goes with it https://github.com/directus/api/pull/1280.

TL;DR that's exactly what we're doing.


Also FYI, the app is supposed to keep you logged in as long as you're using it. Hard to debug why it isn't on your end, but it'll be resolved when we switch to this new auth flow regardless

**Comment by gurumark** [comment]
Where is this change getting implemented? App or API?
I am running custom docker images. I will update the images accordingly. Thanks for your quick action on this issue!!

**Comment by rijkvanzanten** [comment]
It's both! The API will start supporting cookies next to JWTs. The app in turn will start relying on this session cookie instead of having to keep the JWT 'alive', which in turn means that the app can be logged in as long as the user wants. Next to this, we're also implementing a logout endpoint so you can log out of different sessions at will

**Comment by gurumark** [comment]
Is there an ETA for these changes or any specific version I should keep track of? I'd like to be notified if possible.

**Comment by mpfrenette** [comment]
Is there any updates on this issue? It is a major issue on our side.

**Comment by benhaynes** [comment]
Yes, we are actively working on the fix now — it's a full auth flow refactor, so it's taking a little while. It should be released in the next week or so.

**Comment by rijkvanzanten** [comment]
This is fixed in v8

**Comment by gurumark** [comment]
I have upgraded to v.8.3.0 and v.7.11. I still have the same issue. It logs out approx. 5-10 minutes and there is no warning. I tested this in Chrome and Firefox (latest).
This is a critical issue for us. Please advice.

**Comment by rijkvanzanten** [comment]
What do you mean by 

> I have upgraded to v.8.3.0 _and_ v.7.11.

Did you upgrade to 8 or 7?

**Comment by gurumark** [comment]
I have upgraded to v8.3.0 using git pull origin and then git check out v8.3.0. The app it self shows 7.11 in the admin. I don't know which one is valid but I assume the admin one is not correct.

**Comment by rijkvanzanten** [comment]
It sounds like you're still on v7.11 then. 

![image](https://user-images.githubusercontent.com/9141017/70918619-5a3ed980-1fed-11ea-816e-60ff7819771f.png)

**Comment by gurumark** [comment]
How can I upgrade this? I did a git pull origin as states in the docs. I did git checkout v8.3.0 after that to make sure. It still shows 7.11 in the login screen.

**Comment by gurumark** [comment]
I am using one app for multiple apis. I use:
git clone -b build https://github.com/directus/app.git dist
for both the app and api.
There is no clear installation instructions for separate api and app servers.

**Comment by rijkvanzanten** [comment]
That's because that's no longer supported in v8+. The app is required to be served by the API now.

**Comment by gurumark** [comment]
Makes more sense. I wish I've known that before wasting my time to make it work. Anyways, I am testing this new 95 minute auto logout. I will let you know.

**Comment by gurumark** [comment]
It works as expected. Thanks.
